src/modules/auth/hooks/use-auth.ts

```diff
@@ -0,0 +1,48 @@
+import {useCallback, useState} from 'react';
+import {Auth, INITIAL_STORE} from '../stores/auth-store';
+import {useLoginMutation} from '../graphql/mutations/login';
+
+export type Credentials = {
+  email: string;
+  password: string;
+};
+
+export const useAuth = () => {
+  const [login] = useLoginMutation();
+  const [isLoading, setIsLoading] = useState(false);
```

Comment:
Se vc for usar um estado loading custom e não for usar os campos retornados pelo `useMutation`, creio que o `client.mutate` já seja o suficiente pra resolver o seu problema. No entanto eu recomendaria nas próximas tracks vc testar usar o useQuery ou useMutation "de verdade" pra vc ir se acostumando. Qualquer coisa te ajudo.